lib/stats/aggregator/stats-change-per-second.c

```diff
@@ -246,12 +246,13 @@ static void
 _calc_sum(StatsAggregatorCPS *self, CPSLogic *logic, time_t *now)
 {
   gsize diff = _get_count(self) - _get_last_count(logic);
+  gdouble elapsed_time_since_last;
   _set_last_count(logic, _get_count(self));
 
   if (!_is_less_then_duration(self, logic, now))
     {
-      gsize elapsed_time_since_last = (gsize)_calc_sec_between_time(&self->last_add_time, now);
-      diff -= _get_average(logic) * elapsed_time_since_last;
+      elapsed_time_since_last = _calc_sec_between_time(&self->last_add_time, now);
```

Comment:
As far as I understand all invocations of _calc_sec_between_time() cast the return value to gsize.

It's a local, inline function. Why don't we change it to return gsize instead?

Honestly the choice to represent time values as gsize is also strange.

Even an int would be better. A time_t would be the best probably..hmm why does

(A follow-up question why difftime() returns double if its arguments are time_ts is also beyond me
 But it's not the fault of this or)

Reply:
After going through few articles/blogs(attaching for reference), I found that the time_t is measured in seconds, and difftime() is equivalent to arithmetic subtraction, but C standards allows fractional units for time_t therefore, it's preffered to use double as the return type. Though I've made the changes as desired.

- [1](https://en.cppreference.com/w/c/chrono/difftime)
- [2](https://en.cppreference.com/w/c/chrono/time_t)
- [3](https://stackoverflow.com/questions/34457931/why-does-difftime-return-a-double)

Reply:
Sorry, I've assumed that difftime() was our own function. I understand that time_t can also be an alias to double, but syslog-ng's codebase assumes that it is an integer value. If we were to port to one of those platforms where time_t is a double, we would probably have trouble compiling.

But for now, I don't think we want to change this assumption, so this is fine how it is now.